src/cclib/io/wfxwriter.py

```diff
@@ -0,0 +1,475 @@
+        """Rearrange MO related data."""
+        prim_types = self._prim_types()
+        if isinstance(data, numpy.ndarray):
+            data = data.tolist()
+
+        pos_yyx = [key for key, val in enumerate(prim_types)
+                   if val == 17]
+        pos_yyz = [key for key, val in enumerate(prim_types)
+                   if val == 16]
+
+        if pos_yyx:
+            for pos in pos_yyx:
+                data.insert(pos-3, data.pop(pos))
+        if pos_yyz:
+            for pos in pos_yyz:
+                data.insert(pos+3, data.pop(pos + 1))
+
+        return data
+
+
+    def _prim_types(self):
+        """Primitive type numbering starts at 1.
+        The definition of the Cartesian Gaussian primitive types is:
+        1 S, 2 PX, 3 PY, 4 PZ, 5 DXX, 6 DYY, 7 DZZ, 8 DXY, 9 DXZ, 10 DYZ,
+        11 FXXX, 12 FYYY, 13 FZZZ, 14 FXXY, 15 FXXZ, 16 FYYZ, 17 FXYY,
+        18 FXZZ, 19 FYZZ, 20 FXYZ,
+        21 GXXXX, 22 GYYYY, 23 GZZZZ, 24 GXXXY, 25 GXXXZ, 26 GXYYY,
+        27 GYYYZ, 28 GXZZZ,
+        29 GYZZZ, 30 GXXYY, 31 GXXZZ, 32 GYYZZ, 33 GXXYZ, 34 GXYYZ,
+        35 GXYZZ,
+        36 HZZZZZ, 37 HYZZZZ, 38 HYYZZZ, 39 HYYYZZ,
+        40 HYYYYZ, 41 HYYYYY, 42 HXZZZZ, 43 HXYZZZ, 44 HXYYZZ,
+        45 HXYYYZ, 46 HXYYYY, 47 HXXZZZ, 48 HXXYZZ, 49 HXXYYZ,
+        50 HXXYYY, 51 HXXXZZ, 52 HXXXYZ, 53 HXXXYY, 54 HXXXXZ, 55 HXXXXY,
+        56 HXXXXX
+        """
+
+        prim_types = []
+        for atom in self.ccdata.gbasis:
+            for prims in atom:
+                prim_orb = []
+                for i in range(self.num_orb[prims[0]]):
+                    prim_orb += [(self.idx_orb[prims[0]]  + i)]\
+                                * len(prims[1])
+                prim_types += prim_orb
+        return prim_types
+
+    def _prim_types_section(self):
+        prim_types = self._prim_types()
+        # GAMESS specific reordering.
+        if self.ccdata.metadata['package'] == 'GAMESS':
+            prim_types = self._rearrange_modata(prim_types)
+        return self._list_format(prim_types, 10, '%d ')
+
+    def _prim_exps(self):
+        """Space-separated list of primitive exponents."""
+        prim_exps = []
+        for atom in self.ccdata.gbasis:
+            for prims in atom:
+                prim_exps += [prim[0] for prim in prims[1]]\
+                            * self.num_orb[prims[0]]
+        return self._list_format(prim_exps, 5)
+
+    def _mo_occup_nos(self):
+        """MO occupation numbers."""
+        occup = []
+        electrons = self._no_electrons()
+        alpha = self._no_alpha_electrons()
+        beta = self._no_beta_electrons()
+        if len(self.ccdata.homos) == 1:
+            occup += ['{:22.11E}'.format(2)] * int(electrons / 2) +\
+                        ['{:22.11E}'.format(1)] * (electrons % 2)
+        else:
+            occup += ['{:22.11E}'.format(1)] *  + alpha +\
+                        ['{:22.11E}'.format(1)] * beta
+        return occup
+
```

Comment:
Official description of wfx format does not talk about any special treatment for excited states.